test/integration/webhook/webhook.go

```diff
@@ -0,0 +1,171 @@
+// SPDX-FileCopyrightText: 2021 SAP SE or an SAP affiliate company and Gardener contributors.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package webhook
+
+import (
+	"context"
+	"path/filepath"
+
+	"github.com/onsi/ginkgo"
+	"github.com/onsi/gomega"
+
+	admissionregistrationv1 "k8s.io/api/admissionregistration/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+
+	"github.com/gardener/landscaper/apis/config"
+	lsv1alpha1 "github.com/gardener/landscaper/apis/core/v1alpha1"
+	hdv1alpha1 "github.com/gardener/landscaper/apis/deployer/helm/v1alpha1"
+	kutil "github.com/gardener/landscaper/pkg/utils/kubernetes"
+	"github.com/gardener/landscaper/test/framework"
+	"github.com/gardener/landscaper/test/utils"
+)
+
+// RegisterTests registers all tests of the package
+func RegisterTests(f *framework.Framework) {
+	SimpleWebhookTest(f)
+}
+
+func SimpleWebhookTest(f *framework.Framework) {
+	dumper := f.Register()
+
+	_ = ginkgo.Describe("SimpleWebhookTest", func() {
+
+		ginkgo.It("should have created a ValidatingWebhookConfiguration", func() {
+			ctx := context.Background()
+			defer ctx.Done()
+			state, cleanup, err := f.NewState(ctx)
+			utils.ExpectNoError(err)
+			dumper.AddNamespaces(state.Namespace)
+			defer func() {
+				gomega.Expect(cleanup(ctx)).ToNot(gomega.HaveOccurred())
+			}()
```

Comment:
the state is not needed if you create no resources in the test

Reply:
I agree, but if for some reason the webhook doesn't work and allows the creation of the faulty resource - which would be a quite odd case, but possible in theory - we will actually create resources. Should we keep it in for this reason or is it too much of a special case to justify the additional code?

Reply:
but i can disable all webhooks by configuration. So the resources would not be validated but we would still create an empty validation webhook with no rules

Reply:
I'm confused by your comment. That is an argument for keeping the state handling, isn't it? 
If you disable the webhook, the webhook validation tests will fail - because they expect the resources to be denied - and there will be faulty resources in the test cluster.

Reply:
yes sry ignore my first comment.

The issue here is that the cleanup function currently only cleans up resources that are created with `state.create`.
Resources created by the controller are not tracked and currently not cleaned up.
Also the cleanup only removed resources that are created in the test namespace not in the landscaper namespace. (which would have no affect on validation webhooks)

In https://github.com/gardener/landscaper/pull/114 i added such a behavior.
But that cleanup is not triggered by the state cleanup, its triggered automatically.

Reply:
Ah, I got it now, thanks.